addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -225,7 +235,11 @@ func _submit_input(input: Dictionary, tick: int):
 		sanitized[property] = value
 	
 	if sanitized.size() > 0:
-		_inputs[tick] = sanitized
+		for property in sanitized:
+			for i in range(0, sanitized[property].size()):
+				# We received an array of current and previous inputs, merge them into our history.
+				_inputs[tick -i] = PropertySnapshot.merge(_inputs.get(tick - i, {}), {property: sanitized[property][i]})
+
 		_earliest_input = min(_earliest_input, tick)
```

Comment:
I think we could lose some resimulations by not updating `_earliest_input` based on redundancy. On the other hand, if we just did `_earliest_input = min(_earliest_input, tick - redundancy)` or something similar, we would incur a fixed extra cost, even if it's not needed. 

I think the following could work:
```suggestion
		for property in sanitized:
			for i in range(0, sanitized[property].size()):
				var t = tick - i
				var old_input = _inputs.get(t, {}).get(property)
				var new_input = sanitized[property][i]
				
				if old_input != new_input:
					# We received an array of current and previous inputs, merge them into our history.
					_inputs[t] = _inputs.get(t, {})
					_inputs[t][property] = new_input
					_earliest_input = min(_earliest_input, t)
```

It checks if any input values have changed, and if so, it considers that tick for resimulation. Also works on single properties instead of merging snapshots, which might be a tiny bit faster.